### PR TITLE
NBS: Fix large HTTP {get,has}Refs/ requests

### LINKED ISSUE #3589: Sync'ing a large dataset over http results in 400
```
export DB=http://bot.perf.attic.io:8000/p/tester0?access_token=u6xZzU7sQYXdThZKqv2bLl2XGbPnGWRzNRT4hP-r4rCvUfOypoVhQXut_V-WNupFH1iFmKGdUdLO87UbO2Em-qVBucIBz9HrX7oLwBL1VEUlvFDoh9cnGj20e650aY6RyQbmae4dzDFEXkLRdM5s4tRgu6rtqX0zD9LzX6pKBKbUYftzegB25Zu6HVaoPgVbS2v_yQHsyWjh6n8Vk1Cb_A==
noms sync $DB::exiftractor/input nbs:/tmp/testdb::.
        Error Trace:    try.go:99
                    try.go:37
                    http_chunk_store.go:299
                    http_chunk_store.go:172
                    http_chunk_store.go:270
                    asm_amd64.s:2197
    Error:          Unexpected response: Bad Request


goroutine 50142 [running]:
github.com/attic-labs/noms/go/d.Panic(0xbbeb5c, 0x17, 0xc4286f1e70, 0x1, 0x1)
    /var/workspace/src/github.com/attic-labs/noms/go/d/try.go:37 +0x120
github.com/attic-labs/noms/go/datas.(*httpChunkStore).getRefs(0xc420168c60, 0xc42a1a4000)
    /var/workspace/src/github.com/attic-labs/noms/go/datas/http_chunk_store.go:299 +0x841
github.com/attic-labs/noms/go/datas.(*httpChunkStore).(github.com/attic-labs/noms/go/datas.getRefs)-fm(0xc42a1a4000)
    /var/workspace/src/github.com/attic-labs/noms/go/datas/http_chunk_store.go:172 +0x34
github.com/attic-labs/noms/go/datas.(*httpChunkStore).sendReadRequests.func2(0xc42000e028, 0xc4201d2f10, 0xc420168c60)
    /var/workspace/src/github.com/attic-labs/noms/go/datas/http_chunk_store.go:270 +0x5d
created by github.com/attic-labs/noms/go/datas.(*httpChunkStore).sendReadRequests
```

The server doesn't report an error by default.  Re-running with --verbose, the error is: 

```
Compacted table of 2 Kb in 97.040455ms
Returning bad request:
http: POST too large
```

### PR DESCRIPTION
**Pull request by cmasone-attic**
When we added GetMany and HasMany, we didn't realize that requests
could then be larger than the allowable HTTP form size. This patch
makes the body of getRefs and hasRefs be serialized as binary instead,
which addresses this issue and actually makes the request body more
compact.

Fixes #3589

### CODE REVIEW

**Comment by cmasone-attic** [comment]
PTAL

File: go/datas/serialize_hints.go [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+func serializeHashes(w io.Writer, batch chunks.ReadBatch) {
+	err := binary.Write(w, binary.BigEndian, uint32(len(batch))) // 4 billion hashes is probably absurd. Maybe this should be smaller?
+	d.PanicIfError(err)
+	for h := range batch {
+		serializeHash(w, h)
+	}
+}
+
+func serializeHash(w io.Writer, h hash.Hash) {
+	n, err := io.Copy(w, bytes.NewReader(h[:]))
```
Comment (rafael-atticlabs) [inline go/datas/serialize_hints.go]: i think it's safe here to just `w.Write(h[:])`. From the `Writer` go docs: "Write must return a non-nil error if it returns n < len(p)."

File: go/datas/http_chunk_store.go [hunk 2]
```diff
@@ -317,8 +317,9 @@ func (hcs *httpChunkStore) hasRefs(batch chunks.ReadBatch) {
 	u.Path = httprouter.CleanPath(hcs.host.Path + constants.HasRefsPath)
 
 	req := newRequest("POST", hcs.auth, u.String(), buildHashesRequest(batch), http.Header{
-		"Accept-Encoding": {"x-snappy-framed"},
-		"Content-Type":    {"application/x-www-form-urlencoded"},
+		"Accept-Encoding":  {"x-snappy-framed"},
+		"Content-Encoding": {"x-snappy-framed"},
```
Comment (rafael-atticlabs) [inline go/datas/http_chunk_store.go]: seems like snappy wont buy us much here since these are just concat'd hashes.

Comment (cmasone-attic) [inline go/datas/http_chunk_store.go]: done

File: go/datas/serialize_hints.go (hunk 1 above)
Comment (cmasone-attic) [inline go/datas/serialize_hints.go]: done

**Pull request merged**